Source/WebCore/css/CSSBasicShapes.cpp

```diff
@@ -91,9 +91,9 @@ static SerializablePositionOffset buildSerializablePositionOffset(CSSValue* offs
     
     if (side == CSSValueCenter)
         side = defaultSide;
-    else if ((side == CSSValueRight || side == CSSValueBottom) && is<CSSPrimitiveValue>(*amount) && downcast<CSSPrimitiveValue>(*amount).isPercentage()) {
+    else if (auto* primitiveValue = dynamicDowncast<CSSPrimitiveValue>(*amount); (side == CSSValueRight || side == CSSValueBottom) && primitiveValue && primitiveValue->isPercentage()) {
```

Comment:
this allows to keep a one-line, but it changes the order of operation, do we care about performance for things like this here?